docs/user_guides/fs/transformation_functions.md

```diff
@@ -0,0 +1,163 @@
+
+# Transformation Functions    
+
+In AI systems, [transformation functions](https://www.hopsworks.ai/dictionary/transformation) modify data used for machine learning applications, often to improve performance. The [taxonomy of data transformations](../../concepts/mlops/data_transformations.md) introduces three types of data transformation prevalent in all AI systems. Hopsworks offers simple Python APIs to define custom transformation functions. These can be used along with [feature groups](./feature_group/index.md) and [feature views](./feature_view/overview.md) to create [on-demand transformations](./feature_group/on_demand_transformations.md) and [model-dependent transformations](./feature_view/model-dependent-transformations.md), producing modular AI pipelines that are skew-free.
+
+## Creation
+
+User-defined transformation functions can be created in Hopsworks using the [`@udf`](http://docs.hopsworks.ai/hopsworks-api/{{{hopsworks_version}}}/generated/api/udf/) decorator. These functions should be designed as Pandas functions, meaning they must take input features as a [Pandas Series](https://pandas.pydata.org/docs/reference/api/pandas.Series.html) and return either a Pandas Series or a [Pandas DataFrame](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html). Hopsworks automatically executes the defined transformation function as a [`pandas_udf`](https://spark.apache.org/docs/3.1.2/api/python/reference/api/pyspark.sql.functions.pandas_udf.html) in a PySpark application and as Pandas functions in Python clients.
+
+Transformation functions created in Hopsworks can be directly attached to feature views or feature groups or stored in the feature store for later retrieval. These functions can be part of a library [installed](../../user_guides/projects/python/python_install.md) in Hopsworks or be defined in a [Jupyter notebook](../../user_guides/projects/jupyter/python_notebook.md) running a Python kernel or added when starting a Jupyter notebook or [Hopsworks job](../../user_guides/projects/jobs/spark_job.md).
+
+!!! warning "Java/Scala support"
+
+    Creating and attaching Transformation functions to feature views or feature groups are not supported for HSFS Java or Scala client. If feature view/feature groups with transformation function was created using python client, you cannot get training data or get feature vectors from HSFS Java or Scala client.
```

Comment:
Replace HSFS  everywhere with Hopsworks

Reply:
If a feature view or feature group with one or more transformation functions was created using the python client,